cookbooks/prescribed-velocity/prescribed_velocity.cc

```diff
@@ -0,0 +1,204 @@
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/parsed_function.h>
+#include <deal.II/fe/fe_values.h>
+#include <aspect/global.h>
+#include <aspect/simulator_signals.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  // Global variables (to be set by parameters)
+  bool prescribe_internal_velocities;
+
+  // Because we do not initially know what dimension we're in, we need
+  // function parser objects for both 2d and 3d.
+  Functions::ParsedFunction<2> prescribed_velocity_indicator_function_2d (2);
+  Functions::ParsedFunction<3> prescribed_velocity_indicator_function_3d (3);
+  Functions::ParsedFunction<2> prescribed_velocity_function_2d (2);
+  Functions::ParsedFunction<3> prescribed_velocity_function_3d (3);
+
+  /**
+   * Declare additional parameters.
+   */
+  void declare_parameters(const unsigned int dim,
+                          ParameterHandler &prm)
+  {
+    prm.declare_entry ("Prescribe internal velocities", "false",
+                       Patterns::Bool (),
+                       "Whether or not to use any prescribed internal velocities. "
+                       "Values are described by a function in parameter section "
+                       "Prescribed velocities/Function. Values below -1e20 are "
+                       "not prescribed.");
```

Comment:
I think this comment is no longer current. You probably want to say that values should be one or zero and what these indicate.